keras_cv/models/backbones/vit/vit_backbone.py

```diff
@@ -0,0 +1,177 @@
+from keras_cv.layers.vit_layers import PatchingAndEmbedding
+from keras_cv.models import utils
+from keras_cv.models.backbones.backbone import Backbone
+from keras_cv.models.backbones.vit.vit_backbone_presets import backbone_presets
+from keras_cv.models.backbones.vit.vit_backbone_presets import (
+    backbone_presets_with_weights,
+)
+from keras_cv.utils.python_utils import classproperty
+
+
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class ViTBackbone(Backbone):
+    """Instantiates the ViT architecture.
+
+    Reference:
+        - [An Image is Worth 16x16 Words: Transformers for Image Recognition at Scale](https://arxiv.org/abs/2010.11929v2)
+        (ICLR 2021)
+    This function returns a Keras {name} model.
+
+    The naming convention of ViT models follows: ViTSize_Patch-size
+        (i.e. ViTS16).
+    The following sizes were released in the original paper:
+        - S (Small)
+        - B (Base)
+        - L (Large)
+    But subsequent work from the same authors introduced:
+        - Ti (Tiny)
+        - H (Huge)
+
+    The parameter configurations for all of these sizes, at patch sizes 16 and
+    32 are made available, following the naming convention laid out above.
+
+    For transfer learning use cases, make sure to read the
+    [guide to transfer learning & fine-tuning](https://keras.io/guides/transfer_learning/).
+
+    Args:
+        mlp_dim: the dimensionality of the hidden Dense layer in the transformer
+            MLP head
+        include_rescaling: bool, whether to rescale the inputs. If set to
+            True, inputs will be passed through a `Rescaling(1/255.0)` layer.
+        name: string, model name.
+        input_shape: optional shape tuple, defaults to (None, None, 3).
+        input_tensor: optional Keras tensor (i.e. output of `layers.Input()`)
+            to use as image input for the model.
+        project_dim: the latent dimensionality to be projected into in the
+            output of each stacked transformer encoder
+        activation: the activation function to use in the first `layers.Dense`
+            layer in the MLP head of the transformer encoder
+        attention_dropout: the dropout rate to apply to the `MultiHeadAttention`
+            in each transformer encoder
+        mlp_dropout: the dropout rate to apply between `layers.Dense` layers
+            in the MLP head of the transformer encoder
+        num_heads: the number of heads to use in the `MultiHeadAttention` layer
+            of each transformer encoder
+        transformer_layer_num: the number of transformer encoder layers to stack
+            in the Vision Transformer
+        patch_size: the patch size to be supplied to the Patching layer to turn
+            input images into a flattened sequence of patches
+        **kwargs: Pass-through keyword arguments to `keras.Model`.
+    """  # noqa: E501
+
+    def __init__(
+        self,
+        *,
+        include_rescaling=False,
```

Comment:
We don't currently include a default for the base class of other backbones for this parameter, so let's omit it here as well

Reply:
Got it. Removed!